### PR TITLE
Type hint

### PR DESCRIPTION
**Pull request by yinweisu**
Issue #, if available:

Description of changes:

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/autogluon/cloud/predictor/cloud_predictor.py [hunk 1]
```diff
@@ -186,7 +196,7 @@ def generate_trust_relationship_and_iam_policy_file(
 
         return {"trust_relationship": trust_relationship_file_path, "iam_policy": iam_policy_file_path}
 
-    def info(self):
+    def info(self) -> dict:
```
Comment (gradientsky) [inline src/autogluon/cloud/predictor/cloud_predictor.py:199]: if you know types, you can specify types of entries (i.e. `Dict[str, Any]`)

File: src/autogluon/cloud/predictor/cloud_predictor.py [hunk 2]
```diff
@@ -390,20 +400,20 @@ def _upload_fit_artifact(
...
-        framework_version="latest",
-        job_name=None,
-        instance_type="ml.m5.2xlarge",
-        instance_count=1,
-        volume_size=100,
-        custom_image_uri=None,
-        wait=True,
-        autogluon_sagemaker_estimator_kwargs=None,
+        predictor_init_args: dict,
+        predictor_fit_args: dict,
```
Comment (gradientsky) [inline src/autogluon/cloud/predictor/cloud_predictor.py:404]: `Dict` types

File: src/autogluon/cloud/predictor/cloud_predictor.py [hunk 3]
```diff
@@ -622,16 +627,16 @@ def _upload_predictor(self, predictor_path, key_prefix):
...
-        framework_version="latest",
-        instance_type="ml.m5.2xlarge",
-        initial_instance_count=1,
-        custom_image_uri=None,
-        wait=True,
-        model_kwargs=None,
+        predictor_path: Optional[str] = None,
+        endpoint_name: Optional[str] = None,
+        framework_version: Optional[str] = "latest",
+        instance_type: Optional[str] = "ml.m5.2xlarge",
```
Comment (gradientsky) [inline src/autogluon/cloud/predictor/cloud_predictor.py:633]: why this is optional?

File: src/autogluon/cloud/predictor/cloud_predictor.py [hunk 4]
```diff
@@ -622,16 +627,16 @@ def _upload_predictor(self, predictor_path, key_prefix):
...
+        initial_instance_count: Optional[int] = 1,
```
Comment (gradientsky) [inline src/autogluon/cloud/predictor/cloud_predictor.py:634]: Why this is optional?

File: src/autogluon/cloud/predictor/cloud_predictor.py [hunk 5]
```diff
@@ -622,16 +627,16 @@ def _upload_predictor(self, predictor_path, key_prefix):
...
+        custom_image_uri: Optional[str] = None,
+        wait: Optional[bool] = True,
+        model_kwargs: Optional[dict] = None,
```
Comment (gradientsky) [inline src/autogluon/cloud/predictor/cloud_predictor.py:637]: `Dict` types

File: src/autogluon/cloud/predictor/cloud_predictor.py [hunk 6]
```diff
@@ -1026,22 +1035,22 @@ def _predict(
...
-        save_path=None,
-        model_kwargs=None,
-        transformer_kwargs=None,
+        test_data: Union[str, pd.DataFrame],
+        test_data_image_column: Optional[str] = None,
+        predictor_path: Optional[str] = None,
+        framework_version: Optional[str] = "latest",
+        job_name: Optional[str] = None,
+        instance_type: Optional[str] = "ml.m5.2xlarge",
+        instance_count: Optional[int] = 1,
```
Comment (gradientsky) [inline src/autogluon/cloud/predictor/cloud_predictor.py:1044]: Why optional?

File: src/autogluon/cloud/predictor/cloud_predictor.py [hunk 7]
```diff
@@ -1125,23 +1134,23 @@ def predict(
...
-        model_kwargs=None,
-        transformer_kwargs=None,
+        test_data: Union[str, pd.DataFrame],
+        test_data_image_column: Optional[str] = None,
+        include_predict: Optional[bool] = True,
+        predictor_path: Optional[str] = None,
+        framework_version: Optional[str] = "latest",
+        job_name: Optional[str] = None,
+        instance_type: Optional[str] = "ml.m5.2xlarge",
+        instance_count: Optional[int] = 1,
```
Comment (gradientsky) [inline src/autogluon/cloud/predictor/cloud_predictor.py:1144]: Ditto

File: src/autogluon/cloud/predictor/cloud_predictor.py [hunk 8]
```diff
@@ -1330,7 +1339,7 @@ def _download_predictor(self, path, save_path):
         unzip_file(tarball_path, save_path)
         return save_path
 
-    def save(self, silent=False):
+    def save(self, silent: Optional[bool] = False) -> None:
```
Comment (gradientsky) [inline src/autogluon/cloud/predictor/cloud_predictor.py:1342]: why `Optional`? Can it be `None`?

File: src/autogluon/cloud/predictor/multimodal_cloud_predictor.py [hunk 9]
```diff
@@ -50,7 +54,12 @@ def _load_predict_real_time_test_data(self, test_data, test_data_image_column):
 
         return test_data, content_type
 
-    def predict_real_time(self, test_data, test_data_image_column=None, accept="application/x-parquet"):
+    def predict_real_time(
+        self,
+        test_data: Union[str, pd.DataFrame],
+        test_data_image_column: Optional[str] = None,
+        accept: Optional[str] = "application/x-parquet",
```
Comment (gradientsky) [inline src/autogluon/cloud/predictor/multimodal_cloud_predictor.py:61]: why `Optional`? Can it be `None`?

File: src/autogluon/cloud/predictor/multimodal_cloud_predictor.py [hunk 10]
```diff
@@ -91,7 +100,12 @@ def predict_real_time(self, test_data, test_data_image_column=None, accept="appl
 
         return pred
 
-    def predict_proba_real_time(self, test_data, test_data_image_column=None, accept="application/x-parquet"):
+    def predict_proba_real_time(
+        self,
+        test_data: Union[str, pd.DataFrame],
+        test_data_image_column: Optional[str] = None,
+        accept: Optional[str] = "application/x-parquet",
```
Comment (gradientsky) [inline src/autogluon/cloud/predictor/multimodal_cloud_predictor.py:107]: Ditto

File: src/autogluon/cloud/predictor/tabular_cloud_predictor.py [hunk 11]
```diff
@@ -61,7 +65,12 @@ def _load_predict_real_time_test_data(self, test_data, test_data_image_column):
 
         return test_data
 
-    def predict_real_time(self, test_data, test_data_image_column=None, accept="application/x-parquet"):
+    def predict_real_time(
+        self,
+        test_data: Union[str, pd.DataFrame],
+        test_data_image_column: Optional[str] = None,
+        accept: Optional[str] = "application/x-parquet",
```
Comment (gradientsky) [inline src/autogluon/cloud/predictor/tabular_cloud_predictor.py:72]: ditto

File: src/autogluon/cloud/predictor/tabular_cloud_predictor.py [hunk 12]
```diff
@@ -94,7 +103,12 @@ def predict_real_time(self, test_data, test_data_image_column=None, accept="appl
 
         return pred
 
-    def predict_proba_real_time(self, test_data, test_data_image_column=None, accept="application/x-parquet"):
+    def predict_proba_real_time(
+        self,
+        test_data: Union[str, pd.DataFrame],
+        test_data_image_column: Optional[str] = None,
+        accept="application/x-parquet",
```
Comment (gradientsky) [inline src/autogluon/cloud/predictor/tabular_cloud_predictor.py:110]: ditto

File: src/autogluon/cloud/predictor/cloud_predictor.py [hunk 13]
```diff
@@ -60,7 +62,7 @@ class CloudPredictor(ABC):
 
     predictor_file_name = "CloudPredictor.pkl"
 
-    def __init__(self, cloud_output_path, local_output_path=None, verbosity=2):
+    def __init__(self, cloud_output_path: str, local_output_path: Optional[str] = None, verbosity: int = 2) -> None:
```
Comment (tonyhoo) [inline src/autogluon/cloud/predictor/cloud_predictor.py:65]: Can we run local mode without passing `cloud_output_path`

File: src/autogluon/cloud/predictor/cloud_predictor.py [hunk 14]
```diff
@@ -137,7 +145,7 @@ def endpoint_name(self):
     @staticmethod
     def generate_trust_relationship_and_iam_policy_file(
         account_id: str, cloud_output_bucket: str, output_path: Optional[str] = None
-    ):
+    ) -> dict:
```
Comment (tonyhoo) [inline src/autogluon/cloud/predictor/cloud_predictor.py:148]: Dict with more type mapping hints

File: src/autogluon/cloud/predictor/cloud_predictor.py (hunk 13 above)
Comment (yinweisu) [reply src/autogluon/cloud/predictor/cloud_predictor.py:65]: Well, no. The sagemaker training job always output the artifact to s3

File: src/autogluon/cloud/predictor/cloud_predictor.py [hunk 15]
```diff
@@ -114,19 +116,25 @@ def __init__(self, cloud_output_path, local_output_path=None, verbosity=2):
...
 
     @property
     def _realtime_predictor_cls(self):
         return AutoGluonRealtimePredictor
 
     @property
-    def is_fit(self):
+    def is_fit(self) -> bool:
+        """
+        Whether this is CloudPredictor fitted already
```
Comment (gidler) [inline src/autogluon/cloud/predictor/cloud_predictor.py:132]: I think this makes more sense: `Whether this CloudPredictor is fitted already`

File: src/autogluon/cloud/predictor/cloud_predictor.py [hunk 16]
```diff
@@ -390,20 +398,20 @@ def _upload_fit_artifact(
...
-        framework_version="latest",
-        job_name=None,
-        instance_type="ml.m5.2xlarge",
-        instance_count=1,
-        volume_size=100,
-        custom_image_uri=None,
-        wait=True,
-        autogluon_sagemaker_estimator_kwargs=None,
+        predictor_init_args: Dict,
+        predictor_fit_args: Dict,
```
Comment (gidler) [inline src/autogluon/cloud/predictor/cloud_predictor.py:402]: Do we know the types for the keys and values in these dicts?

**Pull request merged**